x-pack/plugins/global_search_bar/public/lib/result_to_option.tsx

```diff
@@ -42,21 +42,19 @@ export const resultToOption = (
       ? [{ text: categoryLabel }]
       : [{ text: cleanMeta((meta.displayName as string) ?? type) }];
 
-  if (getTag && tagIds.length) {
-    const tags = tagIds.map(getTag).filter((tag, index) => {
+  if (tagIds.length && getTagList) {
+    const tags = getTagList().filter((tag, index) => {
       if (!tag) {
-        // eslint-disable-next-line no-console
+        //  eslint-disable-next-line no-console
         console.warn(
           `SearchBar: Tag with id "${tagIds[index]}" not found. Tag "${tagIds[index]}" is referenced by the search result "${result.type}:${result.id}". Skipping displaying the missing tag.`
         );
         return false;
       }
-
       return true;
     }) as Tag[];
 
     if (tags.length) {
-      // TODO #85189 - refactor to use TagList instead of getTag
       option.append = <ResultTagList tags={tags} searchTagIds={searchTagIds} />;
     }
```

Comment:
The `console.warn` will never be called because the `const tags = getTagList().filter()` loops through the known tags. Let's use a `for` loop to go through the `tagIds`, which lets us check for unknown tags:
```typescript
  if (tagIds.length && getTagList) {
    const tagList = getTagList();
    const tags: Tag[] = [];
    for (let index = 0; index < tagIds.length; index++) {
      const tag = tagList.find(({ id: tagId }) => tagId === tagIds[index]);
      if (!tag) {
        // eslint-disable-next-line no-console
        console.warn(
          `SearchBar: Tag with id "${tagIds[index]}" not found. Tag "${tagIds[index]}" is referenced by the search result "${result.type}:${result.id}". Skipping displaying the missing tag.`
        );
      } else {
        tags.push(tag);
      }
    }

    if (tags.length) {
      option.append = <ResultTagList tags={tags} searchTagIds={searchTagIds} />;
    }
  }
```